pkg/apis/mysql/v1alpha1/types.go

```diff
@@ -0,0 +1,400 @@
+	Status BackupStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupList is a list of Backups.
+type BackupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []Backup `json:"items"`
+}
+
+// BackupScheduleSpec defines the specification for a MySQL backup schedule.
+type BackupScheduleSpec struct {
+	// Schedule specifies the cron string used for backup scheduling.
+	Schedule string `json:"schedule"`
+
+	// BackupTemplate is the specification of the backup structure
+	// to get scheduled.
+	BackupTemplate BackupSpec `json:"backupTemplate"`
+}
+
+// BackupSchedulePhase is a string representation of the lifecycle phase
+// of a backup schedule.
+type BackupSchedulePhase string
+
+const (
+	// BackupSchedulePhaseNew means the backup schedule has been created but not
+	// yet processed by the backup schedule controller.
+	BackupSchedulePhaseNew BackupSchedulePhase = "New"
+
+	// BackupSchedulePhaseEnabled means the backup schedule has been validated and
+	// will now be triggering backups according to the schedule spec.
+	BackupSchedulePhaseEnabled BackupSchedulePhase = "Enabled"
+
+	// BackupSchedulePhaseFailedValidation means the backup schedule has failed
+	// the controller's validations and therefore will not trigger backups.
+	BackupSchedulePhaseFailedValidation BackupSchedulePhase = "FailedValidation"
+)
+
+// ScheduleStatus captures the current state of a MySQL backup schedule.
+type ScheduleStatus struct {
+	// Phase is the current phase of the MySQL backup schedule.
+	Phase BackupSchedulePhase `json:"phase"`
+
+	// LastBackup is the last time a Backup was run for this
+	// backup schedule.
+	LastBackup metav1.Time `json:"lastBackup"`
+}
+
+// +genclient
+// +genclient:noStatus
+// +resourceName=mysqlbackupschedules
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupSchedule is a MySQL Operator resource that represents a backup
+// schedule of a MySQL cluster.
+type BackupSchedule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   BackupScheduleSpec `json:"spec"`
+	Status ScheduleStatus     `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupScheduleList is a list of BackupSchedules.
+type BackupScheduleList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []BackupSchedule `json:"items"`
+}
+
+// RestoreSpec defines the specification for a restore of a MySQL backup.
+type RestoreSpec struct {
+	// Cluster is a refeference to the Cluster to which the Restore
+	// belongs.
+	Cluster *corev1.LocalObjectReference `json:"cluster"`
+
+	// Backup is a reference to the Backup object to be restored.
+	Backup *corev1.LocalObjectReference `json:"backup"`
+
+	// AgentScheduled is the agent hostname to run the backup on
+	AgentScheduled string `json:"agentscheduled"`
```

Comment:
SheduledAgent looks more correct to me. Also camel case on the json field. 

Or even AgentName